docs/quadratic_programming.rst

```diff
@@ -0,0 +1,316 @@
+Quadratic programming
+=====================
+
+The different specificities of the solvers are summarized in the table below.  
+The best choice will depend on the usage.  
+  
+- *jit*: the algorithm can used with jit or vmap, on GPU/TPU.
+- *matvec*: the input can be given as matvec instead of dense matrices.
+- *precision*: accuracy expected when the solver succeeds to converge.
+- *stability*: capacity to handle badly scaled problems and matrices with poor conditioning.  
+- *speed*: typical speed on big instances to each its maximum accuracy.
+- *input format*: see subsections below.
+
+.. list-table:: Quadratic Solvers
+   :widths: 45, 15, 20, 15, 15, 15, 22
+   :header-rows: 1
+
+   * - Name
+     - jit
+     - matvec
+     - precision
+     - stability
+     - speed
+     - input format
+   * - :class:`EqualityConstrainedQP`
+     - yes
+     - yes
+     - ++
+     - \+
+     - +++
+     - (Q, c), (A, b)
+   * - :class:`CvxpyQP`
+     - no
+     - no
+     - +++
+     - +++
+     - \+
+     - (Q, c), (A, b), (G, h)
+   * - :class:`OSQP`
+     - yes
+     - yes
+     - \+
+     - ++
+     - ++
+     - (Q, c), (A, b), (G, h)
+   * - :class:`BoxOSQP`
+     - yes
+     - yes
+     - \+
+     - ++
+     - ++
+     - (Q, c), A, (l, u)
+   * - :func:`linear_solve.solve_cg`
+     - yes
+     - yes
+     - +++
+     - +++
+     - +++
+     - Q, c
+
+
+This table is given as rule of thumb only; on some particular instances
+some solvers may behave unexpectedly better than others.
+In front of difficulties, we suggest to test different combinations of
+algorithms, ``maxiter`` and ``tol`` values.  
+  
+.. warning::
+
+  Those algorithms are guaranteed to converge on **convex problems** only.
+  Hence, the matrix :math:`Q` *must* be Positive Semi Definite. 
+
+
+Equality-constrained QPs
+------------------------
+
+The problem takes the form:
+
+.. math::
+
+    \min_{x} \frac{1}{2} x^\top Q x + c^\top x \textrm{ subject to } A x = b
+
+This class is optimized for QPs with equality constraints only: it supports jit, pytrees and matvec.
+
+.. autosummary::
+  :toctree: _autosummary
+
+    jaxopt.EqualityConstrainedQP
+
+Example::
+
+  from jaxopt import EqualityConstrainedQP
+
+  Q = 2 * jnp.array([[2.0, 0.5], [0.5, 1]])
+  c = jnp.array([1.0, 1.0])
+  A = jnp.array([[1.0, 1.0]])
+  b = jnp.array([1.0])
+
+  qp = EqualityConstrainedQP()
+  sol, _ = qp.run(params_obj=(Q, c), params_eq=(A, b)).params
+
+  print(sol.primal)
+  print(sol.dual_eq)
+
+Ill-posed problems
+~~~~~~~~~~~~~~~~~~
+
+This solver is the fastest for well posed problems, but can behave poorly on badly scaled matrices,
+or with redundant constraints.
+
+If the solver struggles to converge, set ``refine_regularization`` and ``refine_maxiter``::
```

Comment:
Maybe mention that it uses iterative refinement and link to the wikipedia article.